addons/mrp/models/mrp_production.py

```diff
@@ -299,6 +301,11 @@ def onchange_picking_type(self):
         self.location_src_id = self.picking_type_id.default_location_src_id.id or location.id
         self.location_dest_id = self.picking_type_id.default_location_dest_id.id or location.id
 
+    @api.onchange('date_planned_start', 'product_id')
+    def onchange_date_planned(self):
+        self.date_planned_finished = Dt.from_string(
```

Comment:
What is wrong with fields.Datetime.from_string in full?

Reply:
nothing at all, just saw it imported as `Dt` in other part so I think that's what you would like.